addons/account/models/account_move.py

```diff
@@ -2761,9 +2761,15 @@ def write(self, vals):
                     posted_move.line_ids._check_tax_lock_date()
 
                 # Hash the move
-                if vals.get('state') == 'posted':
+                if vals.get('is_move_sent') or vals.get('state') == 'posted':
                     self.flush_recordset()  # Ensure that the name is correctly computed before it is used to generate the hash
-                    for move in self.filtered(lambda m: m.restrict_mode_hash_table and not(m.secure_sequence_number or m.inalterable_hash)).sorted(lambda m: (m.date, m.ref or '', m.id)):
+                    filter_operator = 'in' if vals.get('is_move_sent') else 'not in'
+                    filter_domain = [
+                        '&', ('restrict_mode_hash_table', '=', True),
+                        '&', ('move_type', filter_operator, ('out_invoice', 'out_refund')),
+                        '|', ('secure_sequence_number', '=', False), ('inalterable_hash', '=', False),
+                    ]
+                    for move in self.filtered_domain(filter_domain).sorted(lambda m: (m.date, m.ref or '', m.id)):
```

Comment:
`super` has already been called, so you don't need to check `vals` and can use `_get_move_hash_domain(['|', ('secure_sequence_number', '=', False), ('inalterable_hash', '=', False)])`

Also, isn't the condition on `inalterable_hash` and `secure_sequence_number` the same since both need to be set at the same time?

Reply:
> Also, isn't the condition on inalterable_hash and secure_sequence_number the same since both need to be set at the same time?

they are set at the same time indeed, but isn't it theoretically possible that either of them is False and the other one isn't, since they don't depend on each other? :thinking:

Reply:
They are both part of the same mechanism, which should be atomic. One cannot be set without the other one for the hashing to work.